src/store/ducks/opensrp/events/utils.ts

```diff
@@ -0,0 +1,180 @@
+import { Dictionary } from 'lodash';
+import moment from 'moment';
+import {
+  AGE,
+  CASE_CLASSIFICATION_HEADER,
+  CASE_NOTIF_DATE_HEADER,
+  CLASSIFICATION,
+  DIAGNOSIS_DATE,
+  FOCI_OF_INFECTION,
+  FOCI_OF_RESIDENCE,
+  HOUSE_NUMBER,
+  INVESTIGATION_DATE,
+  NAME,
+  PATIENT_NAME,
+  SPECIES,
+  SURNAME,
+} from '../../../../configs/lang';
+
+export type UUID = string;
+export type DateString = string;
+export type TimeStamp = number;
+
+/** describes an event response from the api */
+export interface RawEvent {
+  type: 'Event';
+  dateCreated: TimeStamp;
+  serverVersion: number;
+  identifiers: Dictionary<string>;
+  baseEntityId: UUID;
+  locationId: UUID;
+  eventDate: TimeStamp;
+  eventType: 'Case Details';
+  formSubmissionId: UUID;
+  providerId: string;
+  duration: number;
+  obs: object[];
+  entityType: 'Case Details';
+  details: {
+    id: string;
+    age: string;
+    bfid: string;
+    flag: string;
+    species: string;
+    surname: string;
+    focus_id: UUID;
+    first_name: string;
+    focus_name: string;
+    case_number: string;
+    family_name: string;
+    focus_reason: string;
+    focus_status: string;
+    house_number: string;
+    ep1_create_date: DateString;
+    ep3_create_date: DateString;
+    investigtion_date: DateString;
+    case_classification: string;
+  };
+  version: number;
+  teamId: UUID;
+  _id: UUID;
+  _rev: string;
+}
+
+/** describes the foci information property */
+interface FociInformation {
+  id: UUID;
+  classification: string;
+  name: string;
+}
+
+/** describes the case information property for a n extracted event */
+export interface CaseInformation {
+  caseClassification: string;
+  diagnosisDate: DateString;
+  species: string;
+  notificationDate: DateString;
+  investigationDate: DateString;
+  houseNumber: string;
+  patientName: string;
+  surname: string;
+  age: string;
+}
+
+/** describes the event object after extracting it from rawEvent */
+export interface Event {
+  id: UUID;
+  baseEntityId: UUID;
+  caseNumber: string;
+  caseInformation: CaseInformation;
+  fociInformation: FociInformation;
+  fociInformationTitle: string;
+}
+
+/** lookup table to decide the title of foci information based on events flag details */
+const FociTitleLookup: Dictionary<string> = {
```

Comment:
Should this be exported? Let's also change to `fociTitleLookup` b/c it's not a class or a type.

Reply:
ok